app/main.py

```diff
@@ -1,2 +1,41 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.size = 8
+        self.dict_ = [None for i in range(self.size)]
+
+    def resize(self) -> None:
+        temp_ = [el for el in self.dict_ if el is not None]
+        if len(temp_) > (self.size * (2 / 3)):
+            self.size *= 2
+            self.dict_ = [None for _ in range(self.size)]
+            for el in temp_:
+                self.__setitem__(el[0], el[2])
+
+    def __len__(self) -> int:
+        return len([el for el in self.dict_ if el is not None])
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        hash_ = hash(key)
+        index_ = hash_ % self.size
+        while self.dict_[index_]:
+            if self.dict_[index_] and self.dict_[index_][0] == key:
+                self.dict_[index_] = [key, hash_, value]
+                return
+                self.resize()
+            index_ = (index_ + 1) % self.size
+        self.dict_[index_] = [key, hash_, value]
+        self.resize()
+
+    def __getitem__(self, key: Hashable) -> None:
+        index_ = hash(key) % self.size
+        counter = 0
+        while counter <= (self.size + 1):
```

Comment:
I'm not sure why you wouldn't use a for loop here. If there's a reason, please do tell.

Reply:
thx for all tips, it's beneficial to me))

Reply:
You're welcome. :)
There are a few more things with the new version.